mods/taskbar-disappearing-icon-fix-win11.wh.cpp

```diff
@@ -0,0 +1,647 @@
+        }
+    }
+
+    return nullptr;
+}
+
+// ===========================================================
+
+void HideAndShowIcon(FrameworkElement taskListButtonElement) {
+    if(g_unloading) {
+        return;
+    }
+    
+    auto iconPanelElement =
+        FindChildByName(taskListButtonElement, L"IconPanel");
+    if (!iconPanelElement) {
+        return;
+    }
+
+    auto iconElement =
+        FindChildByName(iconPanelElement, L"Icon")
+            .as<Controls::Image>();
+
+    if(iconElement) {
+        iconElement.Visibility(Visibility::Collapsed);
+        iconElement.Dispatcher().TryRunAsync(winrt::Windows::UI::Core::CoreDispatcherPriority::Low, [iconElement](){
+            iconElement.Visibility(Visibility::Visible);
+        });
+    }
+}
+
+using TaskListButton_Hook2_t = void(WINAPI*)(void* pThis, struct winrt::hstring w);
+TaskListButton_Hook2_t TaskListButton_Hook2_Original;
+void WINAPI TaskListButton_Hook2_Hook(void* pThis, struct winrt::hstring w) {
+    TaskListButton_Hook2_Original(pThis, w);
+
+    Wh_Log(L">Hook2");
+
+    void* taskListButtonIUnknownPtr = (void**)pThis + 3;
+    winrt::Windows::Foundation::IUnknown taskListButtonIUnknown;
+    winrt::copy_from_abi(taskListButtonIUnknown, taskListButtonIUnknownPtr);
+
+    auto taskListButtonElement = taskListButtonIUnknown.as<FrameworkElement>();
+
+    HideAndShowIcon(taskListButtonElement);
+}
+
+// ===========================================================
+
+bool HookTaskbarViewDllSymbols(HMODULE module) {
+    SYMBOL_HOOK symbolHooks[] = {
+        // {
+        //     {
+        //         LR"(private: void __cdecl winrt::Taskbar::implementation::TaskListButton::UpdateVisualStates(void))",
+        //         LR"(private: void __cdecl winrt::Taskbar::implementation::TaskListButton::UpdateVisualStates(void) __ptr64)",
+        //     },
+        //     (void**)&TaskListButton_Hook2_Original,
+        //     (void*)TaskListButton_Hook2_Hook,
+        // },
+        // {
+        //     {
+        //         LR"(public: void __cdecl winrt::Taskbar::implementation::TaskListButton::Icon(struct winrt::Windows::Storage::Streams::IRandomAccessStream))",
+        //         LR"(public: void __cdecl winrt::Taskbar::implementation::TaskListButton::Icon(struct winrt::Windows::Storage::Streams::IRandomAccessStream) __ptr64)",
+        //     },
+        //     (void**)&TaskListButton_Hook2_Original,
+        //     (void*)TaskListButton_Hook2_Hook,
+        // },
+        {
+            {
+                LR"(public: void __cdecl winrt::Taskbar::implementation::TaskListButton::DisplayName(struct winrt::hstring))",
+                LR"(public: void __cdecl winrt::Taskbar::implementation::TaskListButton::DisplayName(struct winrt::hstring) __ptr64)",
+            },
+            (void**)&TaskListButton_Hook2_Original,
+            (void*)TaskListButton_Hook2_Hook,
+        },
+    };
+
+    return HookSymbolsWithOnlineCacheFallback(module, symbolHooks,
```

Comment:
Please `#include <windhawk_utils.h>` and use `WindhawkUtils::HookSymbols` for this.